elastic/apm-aws-lambda
# Make info request timeout configurable

### Summary
This pull request propagates the `ELASTIC_APM_DATA_FORWARDER_TIMEOUT_SECONDS` parameter to the configuration of the reverse proxy tasked with serving APM Server info requests. A error handler is also defined to enable backoff upon transaction failure. Associated tests have also been written in `main_test`.

With this PR, backoff should be enabled after all failed attempts to communicate with the APM Server.

### Changes
- Propagate `ctx` and `config` through `handleInfoRequest`
- Set the `ResponseHeaderTimeout` of the reverse proxy to `ELASTIC_APM_DATA_FORWARDER_TIMEOUT_SECONDS` 
- Define an error handler triggered upon transport failure
- Modify `main_test` accordingly

### How to test
- Set up a slow / unresponsive APM Server
- Setup a basic Python APM-instrumented Lambda function
- Set an large timeout (30+ seconds)
- Upon cold start, the Lambda function execution should last between 3 to 5 seconds, and an error containing `Error querying version from the APM Server` should be logged.
